### PR TITLE
fix(TDI-39389): Gregorian Date Crash Error with Snowflake component in multithread

### PR DESCRIPTION
**Pull request by wwang-talend**
**What is the current behavior?** (You can also link to an open issue here)
https://jira.talendforge.org/browse/TDI-39389

**Please check if the PR fulfills these requirements**

- [x] The commit message follows Talend standard
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features) ?
- [ ] The code coverage on new code >75%
- [ ] The new code does not introduce new technical issues (sonar / eslint)

**What kind of change does this PR introduce?**

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/1191/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/1191/) 
[Sonar results](https://sonar.datapwn.com/overview?id=org.talend.components%3Acomponents%3Awwang-talend%2FTDI-39389)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/1192/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/1192/) 
[Sonar results](https://sonar.datapwn.com/overview?id=org.talend.components%3Acomponents%3Awwang-talend%2FTDI-39389)

File: components/components-snowflake/src/main/java/org/talend/components/snowflake/runtime/Formatter.java [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+        @Override
+        protected SimpleDateFormat initialValue() {
+            return new SimpleDateFormat("yyyy-MM-dd HH:mm:ss.SSSXXX");
+        }
+
+    };
+
+    private SimpleDateFormat timestampFormatter = timestampFormatterLocal.get();
+
+    Object formatTimestampMillis(Object inputValue) {
```
Comment (mbasiuk-talend) [inline components/components-snowflake/src/main/java/org/talend/components/snowflake/runtime/Formatter.java]: Could you please add javadoc for this format methods for better understanding. Thanks.

File: components/components-snowflake/src/main/java/org/talend/components/snowflake/runtime/Formatter.java [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+
+    private SimpleDateFormat timeFormatter = timeFormatterLocal.get();
+
+    {
+        // Time in milliseconds would mean time from midnight. It shouldn't be influenced by timezone differences.
+        // That's why we have to use GMT.
+        timeFormatter.setTimeZone(TimeZone.getTimeZone("GMT"));
+    }
+
+    private static final ThreadLocal<SimpleDateFormat> timestampFormatterLocal = new ThreadLocal<SimpleDateFormat>() {
```
Comment (mbasiuk-talend) [inline components/components-snowflake/src/main/java/org/talend/components/snowflake/runtime/Formatter.java]: Could you please use correct naming for **static final** constants.

File: components/components-snowflake/src/main/java/org/talend/components/snowflake/runtime/Formatter.java [hunk 3]
```diff
@@ -0,0 +1,98 @@
+package org.talend.components.snowflake.runtime;
+
+import java.text.SimpleDateFormat;
+import java.util.Calendar;
+import java.util.Date;
+import java.util.TimeZone;
+
+public class Formatter {
```
Comment (mbasiuk-talend) [inline components/components-snowflake/src/main/java/org/talend/components/snowflake/runtime/Formatter.java]: Please add javadoc for the class definition and add copyright. Thanks.

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/1238/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/1238/) 
[Sonar results](https://sonar.datapwn.com/overview?id=org.talend.components%3Acomponents%3Awwang-talend%2FTDI-39389)

**Pull request merged**